app/tools/update/npm.js

```diff
@@ -0,0 +1,29 @@
+// native modules
+const childProcess = require('child_process');
+// 3rd party modules
+const Promise = require('bluebird');
+const _ = require('lodash');
+// app modules
+const logger = require('../logger');
+
+const exec = Promise.promisify(childProcess.exec);
+
+
+class Npm {
+  static list(execOptions) {
+    const cmd = 'npm list --json --depth=0 --prod';
+    logger.silly('Reading npm packages');
+    const _options = _.defaults({maxBuffer: 1024 * 1024}, execOptions);
+    return exec(cmd, _options).catch((error) => {
+      throw new Error(`npm list fails: ${error.message}`);
+    }).then(stdout => JSON.parse(stdout));
+  }
+  static install(execOptions, options = '--on-optional --only=production') {
+    const cmd = `npm install ${options}`;
```

Comment:
add todo comment: "emit installation stream (stdout/err) to eventbus -> admin room in SIO..." etc